### PR TITLE
Pass client-supplied payload field to EventListener

### PR DESCRIPTION
**Pull request by geraint0923**
Currently we name this field `X-Presto-Client-Info`.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/server/TestResourceUtil.java [hunk 1]
```diff
@@ -56,6 +57,7 @@ public void testCreateSession()
                         .put(PRESTO_SCHEMA, "testSchema")
                         .put(PRESTO_LANGUAGE, "zh-TW")
                         .put(PRESTO_TIME_ZONE, "Asia/Taipei")
+                        .put(PRESTO_USER_CONTEXT, "testUserContext")
```
Comment (raghavsethi) [inline presto-main/src/test/java/com/facebook/presto/server/TestResourceUtil.java]: If this is expected to be JSON, this should be a JSON string.

File: presto-main/src/test/java/com/facebook/presto/server/TestServer.java [hunk 2]
```diff
@@ -103,6 +104,7 @@ public void testQuery()
                 .setHeader(PRESTO_SOURCE, "source")
                 .setHeader(PRESTO_CATALOG, "catalog")
                 .setHeader(PRESTO_SCHEMA, "schema")
+                .setHeader(PRESTO_USER_CONTEXT, "userContext")
```
Comment (raghavsethi) [inline presto-main/src/test/java/com/facebook/presto/server/TestServer.java]: same

**Comment by raghavsethi** [comment]
Any progress on this?

**Comment by geraint0923** [comment]
Addressed the comments, it's called `additional context` now.

File: presto-benchmark-driver/src/main/java/com/facebook/presto/benchmark/driver/BenchmarkDriverOptions.java [hunk 3]
```diff
@@ -90,6 +90,7 @@ public ClientSession getClientSession()
                 parseServer(server),
                 user,
                 "presto-benchmark",
+                "{\"benchmarkField\":\"benchmarkAdditionalContext\"}",
```
Comment (raghavsethi) [inline presto-benchmark-driver/src/main/java/com/facebook/presto/benchmark/driver/BenchmarkDriverOptions.java]: "null" is valid json.

File: presto-cli/src/main/java/com/facebook/presto/cli/PerfTest.java [hunk 4]
```diff
@@ -159,6 +159,7 @@ public ParallelQueryRunner(int maxParallelism, URI server, String catalog, Strin
                         server,
                         "test-" + i,
                         "presto-perf",
+                        "{\"field\":\"additionalContext\"}",
```
Comment (raghavsethi) [inline presto-cli/src/main/java/com/facebook/presto/cli/PerfTest.java]: "null" is valid json.

File: presto-main/src/test/java/com/facebook/presto/server/TestResourceUtil.java [hunk 5]
```diff
@@ -56,6 +57,7 @@ public void testCreateSession()
                         .put(PRESTO_SCHEMA, "testSchema")
                         .put(PRESTO_LANGUAGE, "zh-TW")
                         .put(PRESTO_TIME_ZONE, "Asia/Taipei")
+                        .put(PRESTO_ADDITIONAL_CONTEXT, "{\"field\":\"testAdditionalContext\"}")
```
Comment (raghavsethi) [inline presto-main/src/test/java/com/facebook/presto/server/TestResourceUtil.java]: "null" is valid json.

File: presto-main/src/test/java/com/facebook/presto/server/TestServer.java [hunk 6]
```diff
@@ -103,6 +104,7 @@ public void testQuery()
                 .setHeader(PRESTO_SOURCE, "source")
                 .setHeader(PRESTO_CATALOG, "catalog")
                 .setHeader(PRESTO_SCHEMA, "schema")
+                .setHeader(PRESTO_ADDITIONAL_CONTEXT, "{\"field\":\"additionalContext\"}")
```
Comment (raghavsethi) [inline presto-main/src/test/java/com/facebook/presto/server/TestServer.java]: Here is where this is actually useful as documentation. Use `"{\"additionalContext\":null}"`

File: presto-tests/src/test/java/com/facebook/presto/execution/TestEventListener.java [hunk 7]
```diff
@@ -58,6 +58,7 @@ private void setUp()
                 .setSystemProperty("task_concurrency", "1")
                 .setCatalog("tpch")
                 .setSchema("tiny")
+                .setAdditionalContext("{\"field\":\"testAdditionalContext\"}")
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/TestEventListener.java]: Replace all usages in this file with `"{\"additionalContext\":null}"`

**Comment by geraint0923** [comment]
Addressed the comments.

**Comment by martint** [comment]
We still need to decide on what to call the additional argument.

**Pull request merged**